bundle/edu.gemini.phase2.skeleton.servlet/src/main/scala/edu/gemini/phase2/template/factory/impl/gmos/GmosBase.scala

```diff
@@ -11,4 +12,16 @@ trait GmosBase[B <: SpBlueprint] extends GroupInitializer[B] {
     (f :: fs.toList).map(f => (f, Math.abs(f.getWavelength.toDouble - lambda))).sortBy(_._2).map(_._1).head
   }
 
+  // SET "Custom Mask MDF" = G(N/S)YYYYS(Q/C/DD/SV/LP/FT)XXX-NN
+  //     where:
+  //     (N/S) is the site
+  //     YYYYS is the semester, e.g. 2015A
+  //     (Q/C/DD/SV/LP/FT) is the program type
+  //     XXX is the program number, e.g. 001, or 012, or 123
+  //     NN should be the string "NN" since the mask number is unknown
+  def defaultCustomMaskName(n: ISPNode): String = {
+    // or, in programmer-speak, delete the hyphens and add -NN
+    n.getProgramID.toString.filterNot(_ == '-') + "-NN"
```

Comment:
By convention it happens to be now that the program number is always 3 digits long, but nothing guarantees it.  Maybe it doesn't matter though.